tidy3d/components/geometry.py

```diff
@@ -3280,8 +3289,374 @@ def _surface_area(self, bounds: Bound) -> float:
+            raise ImportError(
+                "Plotting 'CustomSurfaceMeshGeometry' requires 'networkx' and 'rtree'. "
+                "Please install the 'surfacemesh' dependencies. For example: "
+                "pip install 'tidy3d[surfacemesh]'."
+            ) from import_err
+
+    def inside(
+        self, x: np.ndarray[float], y: np.ndarray[float], z: np.ndarray[float]
+    ) -> np.ndarray[bool]:
+        """For input arrays ``x``, ``y``, ``z`` of arbitrary but identical shape, return an array
+        with the same shape which is ``True`` for every point in zip(x, y, z) that is inside the
+        volume of the :class:`Geometry`, and ``False`` otherwise.
+
+        Parameters
+        ----------
+        x : np.ndarray[float]
+            Array of point positions in x direction.
+        y : np.ndarray[float]
+            Array of point positions in y direction.
+        z : np.ndarray[float]
+            Array of point positions in z direction.
+
+        Returns
+        -------
+        np.ndarray[bool]
+            ``True`` for every point that is inside the geometry.
+        """
+
+        arrays = tuple(map(np.array, (x, y, z)))
+        self._ensure_equal_shape(*arrays)
+        inside = np.zeros((arrays[0].size,), dtype=bool)
+        arrays_flat = map(np.ravel, arrays)
+        arrays_stacked = np.stack(tuple(arrays_flat), axis=-1)
+        inside = self.trimesh.contains(arrays_stacked)
+        return inside.reshape(arrays[0].shape)
+
+    def plot(
+        self, x: float = None, y: float = None, z: float = None, ax: Ax = None, **patch_kwargs
+    ) -> Ax:
+        """Plot geometry cross section at single (x,y,z) coordinate.
+
+        Parameters
+        ----------
+        x : float = None
+            Position of plane in x direction, only one of x,y,z can be specified to define plane.
+        y : float = None
+            Position of plane in y direction, only one of x,y,z can be specified to define plane.
+        z : float = None
+            Position of plane in z direction, only one of x,y,z can be specified to define plane.
+        ax : matplotlib.axes._subplots.Axes = None
+            Matplotlib axes to plot on, if not specified, one is created.
+        **patch_kwargs
+            Optional keyword arguments passed to the matplotlib patch plotting of structure.
+            For details on accepted values, refer to
+            `Matplotlib's documentation <https://tinyurl.com/2nf5c2fk>`_.
+
+        Returns
+        -------
+        matplotlib.axes._subplots.Axes
+            The supplied or created matplotlib axes.
+        """
+
+        log.warning(
+            "Plotting a 'CustomSurfaceMeshGeometry' may give inconsistent results "
+            "if the mesh is not unionized. We recommend unionizing all meshes. "
+            "A 'PermittivityMonitor' can be used to check that the mesh is loaded correctly."
+        )
+
+        super().plot(x=x, y=y, z=z, ax=ax, **patch_kwargs)
```

Comment:
I think it might be better to try invoking `Geometry.plot()` directly. Does that work? there's a chance you'd need to pass in `self` as first arg. `super()` can just be a bit confusing to work with.